communication/nats/receiver.go

```diff
@@ -111,17 +131,22 @@ func (receiver *receiverNATS) Respond(consumer communication.RequestConsumer) er
 		}
 	}
 
+	receiver.mu.Lock()
+	defer receiver.mu.Unlock()
+
+	if subs, ok := receiver.subs[requestTopic]; ok && subs.IsValid() {
```

Comment:
might be a nitpick, but it returns a sub, not subs(the s on the end indicates a plural, in my head at least)

Reply:
correct, fixed it.